skimage/transform/_warps.py

```diff
@@ -16,6 +16,54 @@
 )
 
 
+def _preprocess_resize_output_shape(image, output_shape):
+    """Validate resize output shape according to input image.
+
+    Parameters
+    ----------
+    image: ndarray
+        Image to be resized.
+    output_shape: tuple or ndarray
+        Size of the generated output image `(rows, cols[, ...][, dim])`. If
+        `dim` is not provided, the number of channels is preserved.
+
+    Returns
+    -------
+    image: ndarray
+        The input image reshaped if its number of dimensions is not
+        equal to output_shape_length.
+    output_shape: tuple
+        The output image converted to tuple.
+
+    Raises
+    ------
+    ValueError:
+        If output_shape length is smaller than the image number of
+        dimensions
+
+    Notes
+    -----
+    The input image is reshaped if its number of dimensions is not
+    equal to output_shape_length.
+
+    """
+    output_shape = tuple(np.ravel(output_shape))
+    output_ndim = len(output_shape)
+    input_shape = image.shape
+    if output_ndim > image.ndim:
+        # append dimensions to input_shape
+        input_shape += (1, ) * (output_ndim - image.ndim)
+        image = np.reshape(image, input_shape)
+    elif output_ndim == image.ndim - 1:
```

Comment:
I don't think we want to be making `assumptions` here without providing the user an explicit way out of this.

Adding the `multichannel` parameter would go a long way here.

```
elif multichannel and output_ndim == image.ndim - 1
```

Reply:
I simply wrapped all the checks/preprocessing that are already made in `resize` in this function to avoid code duplication in the new `rescale_local_mean` function...

Reply:
@hmaarrfk, do you think then that we need to add the `multichannel` argument to `resize`?

Reply:
I think this is probably a good idea... BUT, anyway, I suggest pushing all this to 0.19, and I intend to make a big push on the `channel_axis` proposal in the coming months.

Reply:
I like the idea of adding support for `channel_axis` in `resize_local_mean` like you did (:pray:), but adding it here in will brake the behaviour of `resize`. This I think must be managed in a separate PR.